### ISSUE TITLE
Variables wrapped in curly brackets, do not use string !

### ISSUE THREAD

**Issue by Zhengqbbb** [jeff-hykin/better-shell-syntax#40]
## The code with a problem is:
```sh
local a="${ccc}"
local b="$ccc"
```

## It looks like: theme : `Vitesse Theme QB`
** screenshot + theme name preferable **
<img width="644" alt="image" src="https://user-images.githubusercontent.com/40693636/217417245-18fd716f-2fb0-4e46-9d8b-52ba0b1c6699.png">

## It should look like:
** description of ideal **
<img width="691" alt="image" src="https://user-images.githubusercontent.com/40693636/217417350-4be140ed-18e9-4b96-af4c-2f8aa13460a2.png">

**Comment by Zhengqbbb** [comment]
Deug for u:

**Comment by Zhengqbbb** [comment]
Debug code:
```sh
function aaa(){
    local O="${O:-$___XX_DEFAULT}"
    local O="${O:-___XX_DEFAULT}"
    local O="${1:?"Please provide xxx"}"
    local O="${1:+"provide=ppp"}"
    local O="${1##"cut"*}"

    local a=1
    local b=1
    echo "$((a+b))"
}
aaa
```

**Comment by jeff-hykin** [comment]
Yeah sadly only basic parameters are supported at the moment. Here's a message on another issue:



> I'll actually upgrade this issue to a much larger issue, which is that the syntax doesn't really support anything beyond the most basic form of [parameter expansion](https://wiki.bash-hackers.org/syntax/pe)
> 
> supported
> 
>     * `$PARAMETER`
> 
>     * `${PARAMETER}`
> 
> 
> 
> not really supported:
> 
>     * Indirection
>       
>       * `${!PARAMETER}`
> 
>     * Case modification
>       
>       * `${PARAMETER^}`
>       * `${PARAMETER^^}`
>       * `${PARAMETER,}`
>       * `${PARAMETER,,}`
>       * `${PARAMETER~}`
>       * `${PARAMETER~~}`
> 
>     * Variable name expansion
>       
>       * `${!PREFIX*}`
>       * `${!PREFIX@}`
> 
>     * Substring removal (also for filename manipulation!)
>       
>       * `${PARAMETER#PATTERN}`
>       * `${PARAMETER##PATTERN}`
>       * `${PARAMETER%PATTERN}`
>       * `${PARAMETER%%PATTERN}`
> 
>     * Search and replace
>       
>       * `${PARAMETER/PATTERN/STRING}`
>       * `${PARAMETER//PATTERN/STRING}`
>       * `${PARAMETER/PATTERN}`
>       * `${PARAMETER//PATTERN}`
> 
>     * String length
>       
>       * `${#PARAMETER}`
> 
>     * Substring expansion
>       
>       * `${PARAMETER:OFFSET}`
>       * `${PARAMETER:OFFSET:LENGTH}`
> 
>     * Use a default value
>       
>       * `${PARAMETER:-WORD}`
>       * `${PARAMETER-WORD}`
> 
>     * Assign a default value
>       
>       * `${PARAMETER:=WORD}`
>       * `${PARAMETER=WORD}`
> 
>     * Use an alternate value
>       
>       * `${PARAMETER:+WORD}`
>       * `${PARAMETER+WORD}`
> 
>     * Display error if null or unset
>       
>       * `${PARAMETER:?WORD}`
>       * `${PARAMETER?WORD}`

**Comment by Zhengqbbb** [comment]
Well, looking forward to your resolve
I desperately need highlight **what is string and variable**. So I rolled back the VSCode version of me and my colleagues.

```sh
echo "https://${HOST}/api/xxx"
```

**Comment by jeff-hykin** [comment]
well, today's your lucky day. I had already been working on parameter expansion from that other issue. See if `v1.4.0` does the trick for you.

If not I published `v0.0.1` is the original syntax, so you can install that version of the extension without downgrading VS Code itself @Zhengqbbb

**Comment by Zhengqbbb** [comment]
Nicccccce Work ! with: [better-shellscript-syntax](https://marketplace.visualstudio.com/items?itemName=jeff-hykin.better-shellscript-syntax)

Looking forward to your resolve !

1. `printf` should be `support function builtin` not just `entity.name.command`
```sh
# test
printf "%s\n" "hello world"
```
